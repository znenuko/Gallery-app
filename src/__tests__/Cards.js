import axios from "axios";
import { render, screen } from "@testing-library/react";
import Cards from "../components/Cards";
import { images } from "../__mocks__/images";

describe("<Container />", () => {
  it("Snapshot principal", () => {
    render(<Cards />);

    expect(screen.getByLabelText("imagenes")).toMatchSnapshot();
  });

  it("Carga contenido de la API", async () => {
    axios.get = jest.fn().mockResolvedValue({ data: images });

    render(<Cards />);

    const res = await screen.findAllByLabelText("img");

    expect(res).toHaveLength(10);
  });
});
