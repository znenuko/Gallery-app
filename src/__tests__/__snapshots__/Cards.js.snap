// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> Snapshot principal 1`] = `
<div
  aria-label="imagenes"
  class="row"
/>
`;
